### ISSUE TITLE
Numpy Dependency Issue When Running `make tests`

### ISSUE THREAD

**Issue by awstanton** [nipy/nipype#3524]
### Summary

Followed the steps to install nipype as specified here: https://nipype.readthedocs.io/en/latest/devel/testing_nipype.html.

Running nipype in venv with Python 3.10 (also tried 3.8 with same results) on Ubuntu 20.04.5 LTS.

The pytest command succeeded with expected results.

`make tests` fails with an error.

### Actual behavior

Output of `make tests`:

rm -rf build
find . -name "*.pyc" | xargs rm -f
find . -name "__pycache__" -type d | xargs rm -rf
find . -name "*.so" | xargs rm -f
find . -name "*.pyd" | xargs rm -f
rm -f tags
rm -rf doc/_build
rm -f .coverage
python setup.py build_ext -i
running build_ext
python -m pytest --doctest-modules nipype
ImportError while loading conftest '<rootDir>nipype/nipype/conftest.py'.
nipype/__init__.py:20: in <module>
    from .utils.config import NipypeConfig
nipype/utils/config.py:19: in <module>
    import numpy as np
nipype-env/lib/python3.10/site-packages/numpy/__init__.py:140: in <module>
    from . import core
nipype-env/lib/python3.10/site-packages/numpy/core/__init__.py:49: in <module>
    raise ImportError(msg)
E   ImportError:
E
E   IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
E
E   Importing the numpy C-extensions failed. This error can happen for
E   many reasons, often due to issues with your setup or how NumPy was
E   installed.
E
E   We have compiled some common reasons and troubleshooting tips at:
E
E       https://numpy.org/devdocs/user/troubleshooting-importerror.html
E
E   Please note and check the following:
E
E     * The Python version is: Python3.10 from "<rootDir>nipype/nipype-env/bin/python"
E     * The NumPy version is: "1.23.3"
E
E   and make sure that they are the versions you expect.
E   Please carefully study the documentation linked above for further help.
E
E   Original error was: No module named 'numpy.core._multiarray_umath'
make: *** [Makefile:59: test-code] Error 4

### Platform details:

{'commit_hash': 'cf264ac35',
 'commit_source': 'repository',
 'networkx_version': '2.8.6',
 'nibabel_version': '4.0.2',
 'nipype_version': '1.8.6.dev0',
 'numpy_version': '1.23.3',
 'pkg_path': <rootDir>/nipype/nipype',
 'scipy_version': '1.9.1',
 'sys_executable': '/usr/bin/python3',
 'sys_platform': 'linux',
 'sys_version': '3.10.7 (main, Sep  7 2022, 15:22:19) [GCC 9.4.0]',
 'traits_version': '6.4.1'}

### Execution environment
- My python environment in virtual environment outside container

**Comment by effigies** [comment]
What happens when you run `python -c 'import numpy'` in that environment? This appears to be a problem with your numpy installation, but it's unclear how this would happen.

For what it's worth, running `make tests` does work for me. The only thing I can see wrong with the `Makefile` is that we run `build_ext`, but we don't have any extensions to build. It should be harmless, though...

**Comment by awstanton** [comment]
Thanks for the reply. `python -c 'import numpy` worked fine and I confirmed that I could use the numpy.array method after importing.

I'll spend some more time getting my environment sorted out and post any relevant updates or observations.

**Comment by awstanton** [comment]
I reinstalled nipype, except instead of running pip install -e .[dev], I ran pip install -e .[all]. And `make tests` worked completely fine. I think that when I installed the first time, I was doing some extra manual steps to install dependencies that did not seem to be satisfied and also had a new version of python I was configuring so somewhere along the line I probably messed up the numpy setup.

Also, in a separate virtual environment, I installed again using pip install -e .[dev] and `make tests` worked completely fine in that case too.

Thus, I'm closing this issue.
